README.md (#24)
alan-turing-institute/clim-recal: README.md

> # Welcome to clim-recal! 

Reviewer: ```suggestion
# Welcome to the `clim-recal` repository! 
```

## Background

Regional climate models (RCMs) contain systematic errors, or biases in their output [1]. Biases arise in RCMs for a number of reasons, such as the assumptions in the general circulation models (GCMs), and in the downscaling process from GCM to RCM [1,2].

> Researchers, policy-makers and other stakeholders wishing to use publicly available RCMs,need to consider a range of bias “correction” methods (sometimes referred to as bias adjustment or recalibration). Bias correction methods offer a means of adjusting the outputs of RCM in a manner that might better reflect future climate change signals whilst preserving the natural/internal variability of climate [2]. 

Reviewer: ```suggestion
Researchers, policy-makers and other stakeholders wishing to use publicly available RCMs need to consider a range of "bias correction” methods (sometimes referred to as "bias adjustment" or "recalibration"). Bias correction methods offer a means of adjusting the outputs of RCM in a manner that might better reflect future climate change signals whilst preserving the natural and internal variability of climate [2]. 
```

The aim of **clim-recal** is therefore to: 

* To provide non-climate scientists with an extensive guide to the application, disadvatanges/advantages and use of BC methods 
* To provide researchers with a collated set of resources for how to technically apply the BC methods, with a framework for open additions 
* To create accessible information on bias adjustment methods for non quantititative researchers and lay-audience stakeholders 

> We are working in partnership with the MetOffice to ensure the propriety of our work (**do we need sign off to say this??**). We're focusing on the UKCP18 suite of products, with the first dataset of focus their ground-breaking local-scale (2.2km) [Convection Permitting Model (CPM)](https://www.metoffice.gov.uk/pub/data/weather/uk/ukcp18/science-reports/UKCP-Convection-permitting-model-projections-report.pdf). 

Reviewer: ```suggestion
We are working in with the Met Office to ensure the propriety, quality, and usability of our work. We're focusing on the UKCP18 suite of products, with the first dataset of focus as their ground-breaking local-scale (2.2km) [Convection Permitting Model (CPM)](https://www.metoffice.gov.uk/pub/data/weather/uk/ukcp18/science-reports/UKCP-Convection-permitting-model-projections-report.pdf). 
```

> ### We love to collaborate!

Reviewer: ```suggestion
### Let's collaborate!
```

> We hope to bring together the extensive work already undertaken by the climate science community and showcase a range of libraries and techniques. If you have suggestions on the repository, or would like to include a new method (see below) or library, please raise an issue or get in touch! **(can we set up a clim-recal@turing email or is that too much? )**

Reviewer: ```suggestion
We hope to bring together the extensive work already undertaken by the climate science community and showcase a range of libraries and techniques. If you have suggestions on the repository, or would like to include a new method (see below) or library, please raise an issue or [get in touch](mailto:rbowyer@turing.ac.uk)! 
```

Our work is however, just like climate data,  intended to be dynamic, and we are in the process of setting up a pipeline for researchers creating new methods of bias correction to be able to submit their methods for inclusion on in the **clim-recal** repository. 

## Code

In this repo we aim to provide examples of how to run the debiasing pipeline starting from the raw data avalaible from the [MET office via CEDA](https://catalogue.ceda.ac.uk/uuid/ad2ac0ddd3f34210b0d6e19bfc335539) to the creation of debiased (bias corrected) datasets for different time periods. The pipeline has the following steps:

1. Reproject the [UKCP](https://data.ceda.ac.uk/badc/ukcp18/data/land-cpm/uk/2.2km) control and scenario data to the same coordinate system as the [HADs](https://data.ceda.ac.uk/badc/ukmo-hadobs/data/insitu/MOHC/HadOBS/HadUK-Grid/v1.1.0.0/1km) observational data (British National Grid).
2. Resample the HADs data from 1km to 2.2km grid to match the UKCP reprojected grid.

> 3. Run debiasing method on the control and observational data and project it into the scenario dataset. 

Reviewer 2: I think it would be good if we were consistent with language (ie either 'diabiasing' or 'bias correction' - my preference would be the latter just because that is most commonly used in the literature but happy to be convinced otherwise!

### Python

In the `python` subdirectory you can find code for the different steps data processing and debiasing:
    - **Resampling** for the HADsUK datasets from 1km to a 2.2 km grid to match the UKCP re-projected grid.
    - **Data loaders** functions for loading and concatenating data into a single xarray which can be used for running debiasing methods.

>     - **Debiasing scripts** that interface with implementations of the debiasing (bias correction) methods implemented by different libraries (by March 2023 we have only implemented the python-cmethods library).

Reviewer 2: Also see this comment I added to the previous commit - welcome all's thoughts! https://github.com/alan-turing-institute/clim-recal/pull/24#discussion_r1153077668
